source/.clang-tidy

```diff
@@ -0,0 +1,7 @@
+---
```

Comment:
Should this go in the top level directory so that things in examples and test will also have these settings applied? Or do they already with this location?